app/code/ZP/LoyaltyProgram/Controller/Adminhtml/Program/Save.php

```diff
@@ -0,0 +1,171 @@
+            if (empty($data[LoyaltyProgram::PROGRAM_ID])) {
+                $data[LoyaltyProgram::PROGRAM_ID] = null;
+            } else {
+                $programId = (int)$data[LoyaltyProgram::PROGRAM_ID];
+                $data[LoyaltyProgram::PROGRAM_ID] = $programId;
+            }
+
+            if (!$this->checkCorrectProgramId($data[LoyaltyProgram::PROGRAM_ID], $programId)) {
+                $this->messageManager->addErrorMessage('You can\'t edit Basic Programs.');
+
+                return $resultRedirect->setPath('*/*/');
+            }
+
+            $data = $this->prepareDataToSave($data);
+
+            /** @var LoyaltyProgram $program */
+            $program = $this->programFactory->create();
+
+            if ($programId && (is_numeric($programId) && !is_float($programId))) {
+                try {
+                    $program = $this->programRepository->get((int)$programId);
+                } catch (LocalizedException $e) {
+                    $this->messageManager->addErrorMessage(__('This Program no longer exists.'));
+
+                    return $resultRedirect->setPath('*/*/');
+                }
+            }
+
+            $program->setData($data);
+
+            try {
+                $this->programRepository->save($program);
+                $this->messageManager->addSuccessMessage(__('You saved the Program.'));
+            } catch (LocalizedException $e) {
+                $this->messageManager->addErrorMessage($e->getMessage());
+                return $resultRedirect->setPath('*/*/new');
+            } catch (\Exception $e) {
+                $this->messageManager->addExceptionMessage($e, __('Something went wrong while saving the Program.'));
+            }
+
+            if ($programId === null) {
+                $programId = $program->getId();
+            }
+
+            if ($this->getRequest()->getParam('back')) {
+                return $resultRedirect->setPath('*/*/edit', [LoyaltyProgram::PROGRAM_ID => (int)$programId]);
+            }
+
+            return $resultRedirect->setPath('*/*/');
+        }
+
+        return $resultRedirect->setPath('*/*/');
+    }
+
+    private function prepareDataToSave(array $data): array
+    {
+        if (empty($data[LoyaltyProgram::PROGRAM_ID])) {
+            $data[LoyaltyProgram::PROGRAM_ID] = null;
+        } else {
+            $programId = (int)$data[LoyaltyProgram::PROGRAM_ID];
+            $data[LoyaltyProgram::PROGRAM_ID] = $programId;
+        }
+
+        if (isset($data[LoyaltyProgram::IS_ACTIVE]) && $data[LoyaltyProgram::IS_ACTIVE] === '1') {
+            $data[LoyaltyProgram::IS_ACTIVE] = LoyaltyProgram::ACTIVE;
+        }
+
+        if (isset($data[LoyaltyProgram::IS_PROGRAM_MINIMUM])) {
+            if ($data[LoyaltyProgram::IS_PROGRAM_MINIMUM] === '1') {
+                $data[LoyaltyProgram::PREVIOUS_PROGRAM] = BasicProgramsConfig::PROGRAM_MIN;
+            } else {
+                if ($data[LoyaltyProgram::PREVIOUS_PROGRAM] === '0') {
```

Comment:
first moment: why didn't use simply used elseif here?
second one: why don't you just create a separate function something like:
`private function processProgramData(array &$data, string $programKey, string $programConfig, string $programNextOrPrevious): void
{
    if (isset($data[$programKey])) {
        if ($data[$programKey] === '1') {
            $data[$programNextOrPrevious] = $programConfig;
        } elseif ($data[$programNextOrPrevious] === '0') {
            $data[$programNextOrPrevious] = null;
        }

        unset($data[$programKey]);
    }
}

// Usage
$this->processProgramData($data, LoyaltyProgram::IS_PROGRAM_MINIMUM, BasicProgramsConfig::PROGRAM_MIN, LoyaltyProgram::PREVIOUS_PROGRAM);
$this->processProgramData($data, LoyaltyProgram::IS_PROGRAM_MAXIMUM, BasicProgramsConfig::PROGRAM_MAX, LoyaltyProgram::NEXT_PROGRAM);
`

Reply:
use class property